packages/client/src/publish/MessageFactory.ts

```diff
@@ -29,13 +30,19 @@ export const createSignedMessage = async <T>(
     opts: Omit<StreamMessageOptions<T>, 'signature' | 'signatureType' | 'content'> 
     & { serializedContent: string, authentication: Authentication }
 ): Promise<StreamMessage<T>> => {
-    const msg = new StreamMessage<T>({
+    const signature = await opts.authentication.createMessagePayloadSignature(createSignaturePayload({
```

Comment:
```suggestion
    const signature = await opts.authentication.createMessagePayloadSignature(createSignaturePayload({
```

Maybe rethink name `createMessagePayloadSignature`?